### PR TITLE
feat: Move B2bStorefrontModule recipe to the new @spartacus/setup library

### LINKED ISSUE #8536: Move B2bStorefrontModule recipe to the new @spartacus/setup library
In #8423 the `B2bStorefrontModule` should be temporarily moved to the my-account feature lib.
But it's not a good final destination of the b2b recipe. It should be moved to the new `@spartacus/setup` library in a separate entry point `@spartacus/setup/recipes/b2b`

See the planned new libraries structure https://github.com/SAP/spartacus/issues/7959

blocked by #8423

### PR DESCRIPTION
**Pull request by dunqan**
See the planned new libraries structure #7959

Closes #8536

### CODE REVIEW

File: core-libs/setup/recipes/b2b/b2b-storefront.module.ts [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-import { OrganizationModule } from '../organization.module';
+import { OrganizationModule } from '@spartacus/my-account/organization';
 import { defaultB2bOccConfig } from './config/b2b-occ-config';
 import { organizationLayoutConfig } from './config/organization-layout.config';
+import { defaultB2bCheckoutConfig } from './config/b2b-checkout-config';
+
 
 @NgModule({
   imports: [
+    CostCenterModule.forRoot(),
```
Comment (tobi-or-not-tobi) [inline core-libs/setup/recipes/b2b/b2b-storefront.module.ts]: I wonder if we can move this further down in the organization modules, as it's more in context then. i.e in `OrganizationModule`, `OrganizationComponentsModule` or `CostCenterComponentsModule`.

File: core-libs/setup/recipes/b2b/b2b-storefront.module.ts [hunk 2]
```diff
@@ -37,7 +35,14 @@ import { organizationLayoutConfig } from './config/organization-layout.config';
     provideDefaultConfig(defaultB2bOccConfig),
     provideDefaultConfigFactory(defaultCmsContentConfig),
     provideDefaultConfig(defaultB2bCheckoutConfig),
+    provideDefaultConfig({
```
Comment (tobi-or-not-tobi) [inline core-libs/setup/recipes/b2b/b2b-storefront.module.ts]: duplicate, we have this a few lines up as well.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

**Review by marlass — DISMISSED** [review]

**Review by psmul — DISMISSED** [review]

File: core-libs/setup/recipes/b2b/b2b-storefront.module.ts (hunk 2 above)
Comment (dunqan) [reply core-libs/setup/recipes/b2b/b2b-storefront.module.ts]: thanks, fixed

File: core-libs/setup/recipes/b2b/b2b-storefront.module.ts (hunk 1 above)
Comment (dunqan) [reply core-libs/setup/recipes/b2b/b2b-storefront.module.ts]: This one is a CostCenter for b2b checkout

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**